components/telemetry-operator/controller/logpipeline/controller.go

```diff
@@ -81,9 +84,42 @@ func NewReconciler(
 func (r *Reconciler) SetupWithManager(mgr ctrl.Manager) error {
 	return ctrl.NewControllerManagedBy(mgr).
 		For(&telemetryv1alpha1.LogPipeline{}).
+		Watches(
+			&source.Kind{Type: &corev1.Secret{}},
+			handler.EnqueueRequestsFromMapFunc(r.enqueueRequests),
+			builder.WithPredicates(predicate.Funcs{
+				CreateFunc: func(event event.CreateEvent) bool {
```

Comment:
Predicate function is pretty intrusive, please check if with this change we can still create a new pipeline or not. In my tests I had some issues that creating a new pipeline was rejected. So I had to do type assertion to verify if incoming object is a secret

Also this change will not cover the case where a user creates a pipeline which is referencing a secret that does not exist yet and would be create later.

Reply:
I think you were using the filter by predicates on the eventhandler registration level in `WithEventFilter` not on the `Watches` registration where events are already filtered by type Secret and then passed on to the predicates function. You are correct, it wasn't working with that setup so we needed to change it.
We tested it thoroughly with the current changes and they seem fine.